deployment/docker/get_filter_messages/filter_msg_retriever.py

```diff
@@ -0,0 +1,91 @@
+# Python Imports
+import argparse
+import logging
+import urllib
+import requests
+import socket
+import time
+from concurrent.futures import ProcessPoolExecutor, as_completed
+from typing import Dict, List, Optional
+
+logging.basicConfig(level=logging.INFO)
+
+
+def resolve_dns(address: str) -> str:
+    start_time = time.time()
+    name, port = address.split(":")
+    ip_address = socket.gethostbyname(name)
+    elapsed = (time.time() - start_time) * 1000
+    logging.info(f"{address} DNS Response took {elapsed} ms")
+    logging.info(f"Talking with {address}, ip address: {ip_address}")
+
+    return f"{ip_address}:{port}"
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(description="Waku filter message retriever")
+    parser.add_argument('-c', '--contentTopic', type=str, help='Content topic', default="/my-app/1/dst/proto")
+    parser.add_argument('-n', '--numNodes', type=int, help='Number of filter nodes to get messages from', default=1)
+
+    return parser.parse_args()
+
+
+def fetch_all_messages(base_url: str, headers: Dict, address: str) -> Optional[List[str]]:
+    response = requests.get(base_url, headers=headers)
+    if response.status_code != 200:
+        logging.error(f"Error fetching data: {response.status_code}")
+        logging.error(response.text)
+        return None
+
+    data = response.json()
+    messages = [message_data['payload'] for message_data in data]
+    logging.info(f"Retrieved {len(messages)} messages from {address}")
+
+    return messages
+
+
+def process_node_messages(address: str, content_topic: str) -> Optional[List[str]]:
+    node_ip = resolve_dns(address)
+    content_topic = urllib.parse.quote(content_topic, safe='')
+    url = f"http://{node_ip}/filter/v2/messages/{content_topic}"
+    logging.debug(f"Query to {url}")
+    headers = {"accept": "text/plain"}
+
+    return fetch_all_messages(url, headers, address)
+
+
+def main():
+    args = parse_args()
+    args_dict = vars(args)
+    logging.info(f"Arguments: {args_dict}")
+
+    hostname = "filter"
+    port = "8645"
+
+    addresses = [f"{hostname}-{i}:{port}" for i in range(args.numNodes)]
+    content_topic = args.contentTopic
+
+    all_messages = []
+    with ProcessPoolExecutor() as executor:
+        futures = [executor.submit(process_node_messages, address, content_topic) for address in addresses]
+
+        for future in as_completed(futures):
+            result = future.result()
+            if result:
+                all_messages.extend(result)
+
+    if len(all_messages) > 1:
+        it = iter(all_messages)
+        len_messages = len(next(it))
+        if not all(len(_) == len_messages for _ in it):
+            print("False")
+        else:
+            print("True")
```

Comment:
I'm a bit confused with the intention of this, could you explain it to me?

Also, I'd suggest only using `_` when the variable is to be "discarded".

Reply:
`all_messages` is a `List[List]`.
What I want to know if all the lists have the same length. So I get the iterator, I check the first `len` to get the first number as a baseline, and then I check if all `len` from the remaining sublists is the same.
If it is not, then we have False, as not all lists have the same `len`.
I use `_` because the variable is indeed discarded.